### DISCUSSION TITLE
sqitch_tool

### DISCUSSION THREAD

**Discussion by brokenpylons** [theory/sqitch#830] [Show and tell]
Hi,

I made a refactoring utility [sqitch_tool](https://github.com/brokenpylons/sqitch_tool), which allows renaming (moving) and removing changes and the associated files in the deploy, verify and revert directories. It basically solves https://github.com/sqitchers/sqitch/issues/290 and https://github.com/sqitchers/sqitch/issues/295.

I mostly made it to scratch my own itch, so it's written in OCaml (it would probably take me more time to learn Perl, than it took me to write this). The implementation uses a character-level recursive descent [parser](https://github.com/brokenpylons/sqitch_tool/blob/5136864c507f3df39b9270f8abcadafb274cd1a4/lib/plan.ml#L198) for parsing `sqitch.plan`, which might be of interest.

**Comment by theory** [comment]
Woah, how fun is that? :-)
